### PR TITLE
Mark/ch1112/build audit log module

### PR DESCRIPTION
**Pull request by marktran**
This PR implements Audit Log functionality.

- Refactored `client` so it's shared between the `AuditLog` and `SSO` classes
- Added common error handling for 400s, 401s, and 422s
- Added `AuditLog` class with a `create_event` method
- Added VCR for testing to assert against real errors
- Updated Yard docs
- Added README docs—these could use some more work

### CODE REVIEW

**Comment by marktran** [comment]
[ch1328]

File: lib/workos/errors.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module WorkOS
+  class WorkOSError < StandardError
+    attr_reader :http_status
+    attr_reader :request_id
+
+    # Initializes a StripeError.
```
Comment (sbauch) [inline lib/workos/errors.rb]:
```suggestion
    # Initializes a WorkOSError.
```

or?

```suggestion
    # :nodoc:
```

Comment (marktran) [reply lib/workos/errors.rb]: Whoops

**Comment by marktran** [comment]
@sbauch Hadn't considered using modules 🤔. Using singleton classes with instance variables and mixins is confusing to me but let's try it.

Can you take another look?

**Comment by marktran** [comment]
Hmm, is Semaphore not hooked into GitHub status checks?

https://workos.semaphoreci.com/projects/workos-ruby

**Comment by marktran** [comment]
> Hmm, is Semaphore not hooked into GitHub status checks?
> 
> https://workos.semaphoreci.com/projects/workos-ruby

Looks like the status checks broke when we renamed the repo and Sorbet has been failing for a while.

**Comment by sbauch** [comment]
Woof, didn't realize that myself.

I think you'll want to run `srb init` to cover the VCR stuff, then it looks like theres some autofixable things.

lmk if you want a hand!

**Comment by marktran** [comment]
@sbauch Alright, got this Sorbet stuff sorted. I had to fix a bunch of Sorbet errors and run `srb rbi update` to get `VCR` working correctly.

I cleared out your last Approval. Can you take another look?

File: lib/workos/client.rb [hunk 2]
```diff
@@ -47,27 +48,35 @@ def user_agent
...
         message = "#{json['message']} (#{errors})"
-        raise InvalidRequestError.new(message, **options)
+        raise InvalidRequestError.new(
+          message: message,
+          http_status: http_status,
+          request_id: response['x-request-id'],
+        )
       end
     end
+    # rubocop:enable Metrics/MethodLength, Metrics/AbcSize
```
Comment (sbauch) [inline lib/workos/client.rb]: ACAB! 👮

**Pull request merged**